### PR TITLE
#160011175 Implement Users Signup Functionality

### PR DESCRIPTION
**Pull request by janetnim**
#### What does this PR do?
Lets a user signup to the application

#### Description of tasks to be completed
Implement functionality for a user can signup to Authors Haven
#### How should this be manually tested?
```sh
$ git clone https://github.com/andela/ah-titans-frontend.git
```
```sh
$ cd ah-titans-frontend
```
```sh
$ git checkout ft-user-signup-160011175
```
>Install the dependencies using the command below.

```sh
$ npm install
```
>Run the application as below.
```sh
$ npm start
```

screen shot 2018-09-05 at 09 40 11

>When a user successfully signs up, he is redirected to home page and prompted with a message to verify his account as can be seen below:

screen shot 2018-09-05 at 09 44 17

#### What are the relevant Pivotal Tracker Stories?
[#160011175](https://www.pivotaltracker.com/story/show/160011175)

### CODE REVIEW

**Comment by dcDalin** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  :exclamation: Big PR
  </td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:
  - `./__tests__/.eslintrc.js`
- `src/Home/components/__tests__/index.js`
- `src/Home/__tests__/homepage.test.js`
- `src/Login/__tests__/login.test.js`
- `src/Signup/__tests__/Signup.test.js`
- `src/Signup/components/__tests__/SignupForm.js`
- `src/Signup/containers/__tests__/Signup.js`
- `src/reducers/__tests__/exampleReducer.js`
- `src/reducers/__tests__/index.js`
- `src/__tests__/setupEnzyme.js`
- `./__tests__/webpack.common.js`
- `./__tests__/webpack.dev.js`
  If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/Home/components/index.js [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
-    <h1>Welcome to Authors  Haven!</h1>
-  </div>
-);
+const Home = (props) => {
+  const { new_user, } = props.home;
+  return (
+    <div className="app">
+      <Loader />
+      {new_user
+        ? 'You have successfully signed up to Authors Haven!! Please check your email to verify your account'
```
Comment (jmnyarega) [inline src/Home/components/index.js]: we can make use of `react-material-snackbar` here

File: src/Signup/Signup.test.js [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
+      <SignupForm store={store} onChange={() => ''} onClick={() => ''} />,
+    );
   });
 });
 
 describe('Signup from', () => {
   it('renders input fields and a button ', () => {
-    const signupFormWrapper = shallow(<SignupForm />);
+    const signupFormWrapper = shallow(
+      <SignupForm store={store} onChange={() => ''} onClick={() => ''} />,
```
Comment (jmnyarega) [inline src/Signup/Signup.test.js]: We can have this 👉  `<SignupForm store={store} onChange={() => ''} onClick={() => ''} />`  declared only once

File: src/Signup/Signup.test.js [hunk 3]
```diff
@@ -1,21 +1,26 @@
 import React from 'react';
-import { shallow, } from 'enzyme';
+import { shallow } from 'enzyme';
```
Comment (jmnyarega) [inline src/Signup/Signup.test.js]: Lose this empty line

File: src/reducers/signup.test.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+      type: types.SIGNUP_REQUEST,
+    };
+
+    expect(reducer(initialState, startAction)).toEqual({
+      ...initialState,
+      isFetching: true,
+    });
+  });
+
+  // it('should raise an error');
```
Comment (jmnyarega) [inline src/reducers/signup.test.js]: Lose this commented line

File: src/Home/components/index.js (hunk 1 above)
Comment (janetnim) [reply src/Home/components/index.js]: Working on it

**Review by petermwash — APPROVED** [review]

**Review by migot01 — APPROVED** [review]

**Review by dcDalin — APPROVED** [review]

**Pull request merged**